File: backend/app/schemas/absence.py
from pydantic import BaseModel, ConfigDict
from datetime import date
from typing import List

# Importa os schemas que serão aninhados na resposta
from .absence_status import AbsenceStatus
from .absence_type import AbsenceType
from .employee import EmployeeInAbsence


class AbsenceBase(BaseModel):
    """Schema base interno, não diretamente exposto na API de criação."""
    day: date
    employee_id: int
    absence_type_id: int


class AbsenceCreate(AbsenceBase):
    """Schema para a função CRUD criar uma única ausência na DB."""
    status_id: int


class AbsenceRequestCreate(BaseModel):
    """Schema para o cliente submeter um pedido de ausência para um ou mais dias."""
    employee_id: int
    absence_type_id: int
    days: List[date]
    status_id: int = 2  # Status padrão "Pendente"


class AbsenceUpdate(BaseModel):
    """Schema para atualizar o status de uma ausência (aprovar/rejeitar)."""
    status_id: int


class Absence(BaseModel):
    """Schema completo para retornar dados de ausência na API (com nesting)."""
    id: int
    day: date
    status: AbsenceStatus
    absence_type: AbsenceType
    employee: EmployeeInAbsence

    model_config = ConfigDict(from_attributes=True)

File: frontend/management_app/src/interfaces/Absence.ts
// Interface para ausência
export interface Absence {
  id: number;
  day: string;
  status: { id: number; name: string };
  absence_type: { id: number; name: string };
  employee: { id: number; name: string; email: string };
}

// Interface para criar pedido de ausência
export interface AbsenceRequestCreate {
  employee_id: number;
  absence_type_id: number;
  status_id?: number;
  days: string[];
}

// Interface para tipo de ausência
export interface AbsenceType {
  id: number;
  name: string;
}

// Interface para status de ausência
export interface AbsenceStatus {
  id: number;
  name: string;
}